crates/tx5-go-pion-sys/build.rs

```diff
@@ -1,6 +1,83 @@
 use std::process::Command;
 
+#[derive(Debug)]
+enum LinkType {
+    Dynamic,
+    Static,
+}
+
+#[derive(Debug)]
+struct Target {
+    pub go_arch: Option<&'static str>,
+    pub go_os: Option<&'static str>,
+    pub link_type: LinkType,
+}
+
+impl Default for Target {
+    fn default() -> Self {
+        let target_arch = std::env::var("CARGO_CFG_TARGET_ARCH").unwrap();
+
+        let go_arch = match target_arch.as_str() {
+            "arm" => Some("arm"),
+            "aarch64" => Some("arm64"),
+            "x86_64" => Some("amd64"),
+            "x86" => Some("386"),
+            _ => None,
+        };
+
+        let target_os = std::env::var("CARGO_CFG_TARGET_OS").unwrap();
+
+        let go_os = match target_os.as_str() {
+            "windows" => Some("windows"),
+            "macos" => Some("darwin"),
+            "ios" => Some("ios"),
+            "linux" => Some("linux"),
+            "android" => Some("android"),
+            "dragonfly" => Some("dragonfly"),
+            "freebsd" => Some("freebsd"),
+            "openbsd" => Some("openbsd"),
+            "netbsd" => Some("netbsd"),
+            _ => None,
+        };
+
+        let mut link_type = match target_os.as_str() {
+            "windows" | "android" => LinkType::Dynamic,
```

Comment:
Yeah, alas, golang doesn't even support static linking on windows or android, which is why they must be dynamic. Static would be preferable everywhere otherwise.

Reply:
Technically, it supports it on windows, it just isn't compatible with rust.